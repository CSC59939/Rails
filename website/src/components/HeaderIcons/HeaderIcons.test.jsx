import { HeaderIcons } from '..';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Button, Card, Icon } from 'antd';
import { IconButton } from '..';

configure({ adapter: new Adapter() });

const sampleFunction = () => {
  console.log('Button Circle Clicked')
}
describe("SampleComponent", () => {
  it('Expect SampleComponent to return a div with its name', () => { // eslint-disable-line no-undef
    expect(renderer.create(<HeaderIcons />).toJSON()).toMatchSnapshot();
  });
//   it('should contains a IconButton tag with a onclick function which returns Button Circle Clicked', () => {
//     const wrapper = shallow(<HeaderIcons/>);
//     wrapper.find('.add_circle').simulate('click'); // this is the fucntion for simulate the onclick function, but it only runs single node.
// });
  it('should have a class name called IconButton', () => {
   expect(shallow(<HeaderIcons />).exists('.HeaderIcons')).toBe(true)
 });
})
