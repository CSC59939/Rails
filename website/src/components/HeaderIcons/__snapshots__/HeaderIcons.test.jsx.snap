// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SampleComponent Expect SampleComponent to return a div with its name 1`] = `
<div
  className="HeaderIcons"
>
  <button
    className="IconButton"
    onClick={[Function]}
    type="button"
  >
    <i
      className="material-icons"
      style={null}
    >
      add_circle
    </i>
  </button>
  <button
    className="IconButton"
    onClick={[Function]}
    type="button"
  >
    <i
      className="material-icons"
      style={null}
    >
      add_alert
    </i>
  </button>
  <button
    className="IconButton"
    onClick={[Function]}
    type="button"
  >
    <i
      className="material-icons"
      style={null}
    >
      notifications_active
    </i>
  </button>
  <button
    className="IconButton"
    onClick={[Function]}
    type="button"
  >
    <i
      className="material-icons"
      style={null}
    >
      settings
    </i>
  </button>
</div>
`;

exports[`SampleComponent Expect SampleComponent to return a div with its name 2`] = `
<div
  className="HeaderIcons"
>
  <button
    className="IconButton"
    onClick={[Function]}
    type="button"
  >
    <i
      className="material-icons"
      style={null}
    >
      add_circle
    </i>
  </button>
  <button
    className="IconButton"
    onClick={[Function]}
    type="button"
  >
    <i
      className="material-icons"
      style={null}
    >
      add_alert
    </i>
  </button>
  <button
    className="IconButton"
    onClick={[Function]}
    type="button"
  >
    <i
      className="material-icons"
      style={null}
    >
      notifications_active
    </i>
  </button>
  <button
    className="IconButton"
    onClick={[Function]}
    type="button"
  >
    <i
      className="material-icons"
      style={null}
    >
      settings
    </i>
  </button>
</div>
`;
